packages/loader/container-loader/src/container.ts

```diff
@@ -908,11 +909,13 @@ export class Container
 	}
 
 	public dispose(error?: ICriticalContainerError) {
+		this.switchFromEstablishingConnectionToDisconnect("ContainerDisposed");
```

Comment:
I'd recommend to change order. First we lose connection and move to mode "trying to reconnect". Then we move to "not trying to reconnect". That way we always have no active connection when we move to "not trying to reconnect" - fewer states and state transitions to manage.